modules/json/src/smithy4s/json/internals/SchemaVisitorJCodec.scala

```diff
@@ -1040,27 +1064,39 @@ private[smithy4s] class SchemaVisitorJCodec(
   )(dispatch: Alt.Dispatcher[U]): JCodec[U] =
     new TaggedUnionJCodec[U](alternatives)(dispatch) {
 
-      def decodeValue(cursor: Cursor, in: JsonReader): U =
+      def decodeValue(cursor: Cursor, in: JsonReader): U = {
```

Comment:
I'd rather we didn't change the existing implementation but rather created a new one specifically for the case where `JsonUnknown` is present

Reply:
Although, maybe not ...

Reply:
sticking to a shared impl for the time being... Especially in the context of tagged unions I don't expect there to be a lot of overhead, the unhappy path has one extra `else if` but that should be it?